### PR TITLE
Small Energy update

### PR DESCRIPTION
**Pull request by XSlayer300**
## About The Pull Request
Adds a little assortment of things related to energy weapons. It adds a deathsquad-only Pulse Rifle (like we've always seen on SS13, using sprites from TG/Bay), rechargable batteries (directly ported from Bay) and adds some more fluff to the M43's lens, improving its description for a bit. I also added new energy projectile sprites too.

## Why It's Good For The Game
It benefits coders for having recharable batteries, along with other assortments of nice touches.

## Changelog
:cl:
tweak: Made batteries self-rechargable, only the pulse energy rifle's batteries can be recharged.
tweak: Changed the sprites of the laser bolts.
tweak: Changed the lens descriptions, adding more fluff.
add: Added the M19C4 pulse energy rifle, an admin-only deathsquad weapon. Feast your eyes on destruction!
/:cl:

### CODE REVIEW

File: code/modules/projectiles/updated_projectiles/guns/energy.dm [hunk 1]
```diff
@@ -365,3 +365,46 @@
...
+
+/obj/item/weapon/gun/energy/lasgun/pulse
+	name = "\improper M19C4 pulse energy rifle"
+	desc = "A heavy-duty, multifaceted energy weapon that uses pulse-based beam generation technology to emit powerful laser blasts. Because of its complexity and cost, it is rarely seen in use except by specialists and front-line combat personnel. This is a testing model issued only for Asset Protection."
+	force = 23 //Slightly more heftier than the M43, but without the stock.
+	icon_state = "m19c4"
+	item_state = "m19c4"
+	max_shots = 100//codex stuff
+	load_method = CELL //codex stuff
+	ammo = /datum/ammo/energy/lasgun/pulse
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/guns/energy.dm]: this is undefined

File: code/game/objects/items/power_cells.dm [hunk 2]
```diff
@@ -13,6 +13,10 @@
 	var/maxcharge = 1000
 	var/rigged = 0		// true if rigged to explode
 	var/minor_fault = 0 //If not 100% reliable, it will build up faults.
+	var/self_recharge = FALSE // If true, the cell will recharge itself.
```
Comment (spookydonut) [inline code/game/objects/items/power_cells.dm]: nothing in this PR makes use of self_recharge

File: code/modules/projectiles/updated_projectiles/guns/energy.dm [hunk 3]
```diff
@@ -365,3 +365,46 @@
...
+		var/remaining = CEILING((cell.charge / max(cell.maxcharge, 1)) * 100, 25)
+		icon_state = "[base_gun_icon]_[remaining]"
+		item_state = "m19c4_[remaining][flags_item & WIELDED ? "_w" : ""]"
+
+	if(ishuman(user))
+		var/mob/living/carbon/human/M = user
+		if(src == M.l_hand)
+			M.update_inv_l_hand()
+		else if (src == M.r_hand)
+			M.update_inv_r_hand()
```
Comment (Rohesie) [inline code/modules/projectiles/updated_projectiles/guns/energy.dm]:
```suggestion
			M.update_inv_r_hand()

```

File: code/game/objects/items/power_cells.dm (hunk 2 above)
Comment (XSlayer300) [reply code/game/objects/items/power_cells.dm]: Made the pulse magazine able to self_recharge.

File: code/modules/projectiles/updated_projectiles/guns/energy.dm (hunk 1 above)
Comment (XSlayer300) [reply code/modules/projectiles/updated_projectiles/guns/energy.dm]: Added an inhand sprite, as well as ammo fix.

File: code/modules/projectiles/updated_projectiles/magazines/energy.dm [hunk 4]
```diff
@@ -37,3 +37,13 @@
...
+/obj/item/cell/lasgun/pulse
+	name = "\improper M19C4 pulse battery"
+	desc = "An advanced, ultrahigh capacity battery used to power the M19C4 pulse rifle; Uses pulse-based energy rather than laser energy, massively increasing its firepower. It can also recharge on its own."
+	base_ammo_icon = "m43"
+	icon_state = "m43"
+	gun_type = /obj/item/weapon/gun/energy/lasgun/pulse
+	maxcharge = 2000 // 100 shots.
+	self_recharge = TRUE
+	charge_amount = 25 // 10%, 1 shot
+	charge_delay = 20 // Every two seconds.
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/magazines/energy.dm]:
```suggestion
	charge_delay = 20 // Every two seconds.

```

File: code/modules/projectiles/updated_projectiles/magazines/energy.dm [hunk 5]
```diff
@@ -37,3 +37,13 @@
...
```
Comment (spookydonut) [inline code/modules/projectiles/updated_projectiles/magazines/energy.dm]:
```suggestion
	charge_delay = 2 SECONDS
```

Comment (XSlayer300) [reply code/modules/projectiles/updated_projectiles/magazines/energy.dm]: Done, m8.

**Pull request merged**